cbor/record.go

```diff
@@ -101,10 +101,13 @@ func RecordToProto(ctx context.Context, dag format.DAGService, rec thread.Record
 	if err != nil {
 		return nil, err
 	}
-	event, err := EventFromNode(block)
+	event, ok := block.(*Event)
+	if !ok {
+	event, err = EventFromNode(block)
```

Comment:
Is this reasonable? Maybe for some reason the `block` already is a decoded `Event`, so to avoid unmarshaling again.

Wouldn't be useful that `Record` interface has a `GetEvent` that returns the underlying event and does whatever necessary to achieve that? It would be useful here and in other parts (that I'd mention with double asterisk (**) in other comments).